lib/gems/pending/util/mount/miq_s3_session.rb

```diff
@@ -0,0 +1,66 @@
+require 'util/mount/miq_generic_mount_session'
+require 'aws-sdk'
+
+class MiqS3Session < MiqGenericMountSession
+  def initialize(log_settings)
+    log_header = "MIQ(#{self.class.name}-initialize)"
+    logger.debug("#{log_header} initialize: log_settings are #{log_settings}")
+    super(log_settings)
+    logger.debug("#{log_header} initialize: @settings are #{@settings}")
+    raise "username is a required value!" if @settings[:username].nil?
+    raise "password is a required value!" if @settings[:password].nil?
+    raise "region is a required value!" if @settings[:region].nil?
+    @s3 = s3
+  end
+
+  def s3
+    @s3 ||= Aws::S3::Resource.new(:region => @settings[:region], :access_key_id => @settings[:username], :secret_access_key => @settings[:password])
+  end
+
+  def connect
+    @host       = URI(uri).host
```

Comment:
Why even bother having content here? I think it would be more clear to set `@host` and `@mount_path` on initialize and  leave this method with no content with a comment that explains that you don't really have to "connect" to the object store.